src/middlewared/middlewared/alert/service/telegram.py

```diff
@@ -0,0 +1,26 @@
+import requests
+
+from middlewared.alert.base import ThreadedAlertService
+from middlewared.schema import Dict, Str
+
+
+class TelegramAlertService(ThreadedAlertService):
+    title = "Telegram"
+
+    schema = Dict(
+        "Telegram_attributes",
+        Str("base_url", default="https://api.telegram.org/bot"),
+        Str("TELEGRAM_CHAT_ID", required=True, empty=False),
+        Str("TELEGRAM_BOT_TOKEN", required=True, empty=False),
+        strict=True,
+    )
+
+    def send_sync(self, alerts, gone_alerts, new_alerts):
+        base_url = self.attributes["base_url"] or "https://api.Telegram.com/bot"
+	url = {base_url} + {self.attributes['TELEGRAM_BOT_TOKEN']} + '/sendMessage'
```

Comment:
This is not valid python code.

Reply:
sorry for the mis-indentation, i'll fix, written in a text editor not an IDE :/

Reply:
Its not just indentation. 
 {base_url} is a set, which you're mixing with another set and a string.

Reply:
fixed all syntax errors